lib/pinterest/client/media.rb

```diff
@@ -0,0 +1,66 @@
+module Pinterest
+  class Client
+    module Media
+      class UploadFailed < StandardError; end
+
+      DEFAULT_ASSET_TYPE = 'video'.freeze
+      CONTENT_TYPE = 'application/json'.freeze
+
+      def upload(options = {})
+        upload_id, upload_url, upload_parameters = register_upload(options)
+
+        video_url = options.delete(:video_url)
+        media = open(video_url, 'rb')
+
+        upload_resp = connection.send(:post) do |req|
+          req.path = URI.encode(File.join(upload_url, ''))
+          req.headers['Content-Type'] = "multipart/form-data"
+          req.body = upload_parameters.merge(
+            file: Faraday::UploadIO.new(media, content_type(media))
+          )
+        end
+
+        raise UploadFailed unless upload_resp.success?
+
+        movie = FFMPEG::Movie.new(media.path)
+        screenshot = movie.screenshot("#{upload_filename(media)}-screenshot.jpg")
```

Comment:
My concern here is the accumulation of screenshot files in the base directory.

Reply:
Yeah, it's too bad that this doesn't offer a way to screenshot to some sort of object in memory. I'm also not sure how Heroku will handle this. I wonder if you could write this to `/tmp` instead. My guess is that it will get cleared out daily then when the dynos cycle.

Reply:
I was able to write the screenshots to `tmp/` :)